openmrs-client/src/main/java/org/openmrs/mobile/net/helpers/FindPatientsHelper.java

```
@@ -27,6 +28,12 @@ private FindPatientsHelper() {
     }
 
     public static FindPatientListener createFindPatientListener(String lastQuery, int searchId, FindPatientsSearchActivity caller) {
+        try {
+            URI uri = new URI(null,null,null,lastQuery);
+            lastQuery = uri.toString().substring(1);
+        }catch (Exception e){
+            e.printStackTrace();
```

Never eat exceptions. Report them to the user and log.